eip/EIPS/eip-4337.md

```diff
@@ -530,6 +532,7 @@ The result `SHOULD` be set to the **userOpHash** if and only if the request pass
     * The `data` field SHOULD contain a `minimumStake` and `minimumUnstakeDelay`
   * **code: -32506** - transaction rejected because wallet specified unsupported signature aggregator
     * The `data` field SHOULD contain an `aggregator` value
+  * **code: -32507** - transaction rejected because either the UserOperation signature or the paymaster signature is invalid
```

Comment:
rejected because of wallet signature check failed (or paymaster signature, if the paymaster uses its data as signature)

Reply:
Did you mean -32507?  or did you want this to be merged with -32506

Reply:
no, my mistake in the number, only the text.
the failure is for signature failure. normal signature check is UseOp.signature (which is the user's signature)
there is no "paymaster signature" specific field - the paymaster may put some sig info into its paymasterAndData, as our verifyingPaymaster does.
That's why I suggested the text:

- -32507 - transaction rejected because of wallet signature check failed (or paymaster signature, if the paymaster uses its data as signature)

Reply:
Gotchu -- updated.